### PR TITLE
SISRP-5405 Adding Delete APIs and fixing test failures

### PR DESCRIPTION
**Pull request by ctweney**
https://jira.berkeley.edu/browse/SISRP-5405

Lots of new delete APIs, and some improvements/fixes in failing tests. Bamboo is green!

### CODE REVIEW

File: fixtures/xml/campus_solutions/phone_delete.xml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0"?>
+<PHONE_DELETE_RESPONSE xmlns="http://bcs.is.berkeley.edu">
+  <STATUS>Successfully deleted the contact details for individual 25738808 with the phone type CELL</STATUS>
```
Comment (christianv) [inline fixtures/xml/campus_solutions/phone_delete.xml]: @ctweney do we also have fixtures for when the DELETE fails?

**Comment by christianv** [comment]
@ctweney lgtm, feel free to self merge when travis / bamboo are green.

File: fixtures/xml/campus_solutions/phone_delete.xml (hunk 1 above)
Comment (ctweney) [inline fixtures/xml/campus_solutions/phone_delete.xml]: No, but it's not necessary; the failure message structure is exactly the same.

Comment (christianv) [inline fixtures/xml/campus_solutions/phone_delete.xml]: @ctweney ok, do we get an `errored = true` back when it fails?

Comment (ctweney) [inline fixtures/xml/campus_solutions/phone_delete.xml]: Yes, we should.

**Pull request merged**